capa/ida/plugin/cache.py

```diff
@@ -8,26 +8,18 @@
 
 from __future__ import annotations
 
-import copy
 import itertools
 import collections
 from typing import Set, Dict, List, Tuple, Union, Optional
 
 import capa.engine
-from capa.rules import Rule, Scope, RuleSet
+from capa.rules import Scope, RuleSet
 from capa.engine import FeatureSet, MatchResults
 from capa.features.address import NO_ADDRESS, Address
 from capa.ida.plugin.extractor import CapaExplorerFeatureExtractor
 from capa.features.extractors.base_extractor import BBHandle, InsnHandle, FunctionHandle
 
 
-class CapaExplorerRuleSetCache:
-    def __init__(self, rules: List[Rule]):
-        # capa.rules.Ruleset modifies rules, so we use deepcopy to preserve the original list of rules and our cached list of rules
-        self.rules: List[Rule] = copy.deepcopy(rules)
-        self.ruleset: RuleSet = RuleSet(copy.deepcopy(self.rules))
```

Comment:
in this PR you'll see i've removed this class with preference to a single RuleSet instance that we deepcopy. i didn't find that the list of source rules was needed except for:
  1. counting the number of rules, as they'd show up on disk, which i've moved into a property on RuleSet, and
  2. validating a new rule fits into the ruleset, which i've done by simply iterating the ruleset.rules and filtering subscope rules. 

i think a simpler structure is sufficient; however, if you disagree, open to re-adding.